packages/@aws-cdk/aws-apigateway/lib/base-path-mapping.ts

```diff
@@ -13,6 +13,12 @@ export interface BasePathMappingOptions {
    * is undefined, no additional mappings will be allowed on this domain name.
    */
   readonly basePath?: string;
+
+  /**
+   * The Deployment stage name of API
+   * @default - map to deploymentStage of restApi otherwise stage needs to pass in URL
+   */
+  readonly stage?: string;
```

Comment:
Can we instead pass an instance of `Stage` here?

Reply:
Changed it to Stage type.